pdns/rec_channel.cc

```diff
@@ -54,6 +54,12 @@ int RecursorControlChannel::listen(const string& fname)
   if(bind(d_fd, (sockaddr*)&d_local,sizeof(d_local))<0) 
     throw PDNSException("Unable to bind to controlsocket '"+fname+"': "+stringerror());
 
+  // receive buf should be size of max datagram plus address size
+  int bufsz = 60*1024;
+  setsockopt(d_fd, SOL_SOCKET, SO_SNDBUF, &bufsz, sizeof(bufsz));
+  bufsz = 64*1024;
+  setsockopt(d_fd, SOL_SOCKET, SO_RCVBUF, &bufsz, sizeof(bufsz));
```

Comment:
I'm a bit worried about the fact that this might actually _lower_ the buffer size, that's the case on several Linux boxes I tested, for example. Perhaps we could reuse the `setSocketReceiveBuffer()` and `setSocketSendBuffer()` functions from `pdns_recursor.cc` so that we only increase the size? We should probably lower the severity of the messages from `Error` to `Warning` if we do that, by the way.

Reply:
Agreed, it is better to first check the current size. I'll make a 2nd revision.